sklearn/utils/tests/test_stats.py

```diff
@@ -48,23 +66,35 @@ def test_weighted_percentile_zero_weight_zero_percentile():
     assert approx(score) == 4
 
 
-def test_weighted_median_equal_weights():
+@pytest.mark.parametrize(
+    "array_namespace, device, dtype_name", yield_namespace_device_dtype_combinations()
+)
+def test_weighted_median_equal_weights(array_namespace, device, dtype_name):
+    xp = _array_api_for_tests(array_namespace, device)
     # Checks weighted percentile=0.5 is same as median when weights equal
     rng = np.random.RandomState(0)
     # Odd size as _weighted_percentile takes lower weighted percentile
     x = rng.randint(10, size=11)
-    weights = np.ones(x.shape)
+    x = x.astype(dtype_name, copy=False)
+    x = xp.asarray(x, device=device)
+    weights = xp.ones(x.shape)
 
     median = np.median(x)
     w_median = _weighted_percentile(x, weights)
     assert median == approx(w_median)
 
 
-def test_weighted_median_integer_weights():
+@pytest.mark.parametrize(
+    "array_namespace, device, dtype_name", yield_namespace_device_dtype_combinations()
+)
+def test_weighted_median_integer_weights(array_namespace, device, dtype_name):
+    xp = _array_api_for_tests(array_namespace, device)
     # Checks weighted percentile=0.5 is same as median when manually weight
     # data
     rng = np.random.RandomState(0)
     x = rng.randint(20, size=10)
+    x = x.astype(dtype_name, copy=False)
+    x = xp.asarray(x, device=device)
     weights = rng.choice(5, size=10)
```

Comment:
In case we chose to parametrise this test with Array API namespace-things, we need to also steer the type and the device of weights like in the other tests.
```suggestion
    weights = rng.choice(5, size=10)
    weights = weights.astype(dtype_name, copy=False)
    weights = xp.asarray(weights, device=device)
```
There might be a less wordy way to archive this, though.
Also, as I wrote above, I am not sure if this is really necessary.